pulsar/consumer_singletopic.go

```diff
@@ -0,0 +1,41 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package pulsar
+
+type singleTopicConsumer struct {
```

Comment:
What is the purpose of this?

Reply:
since I can't close message channel directly in internal consumer. So I wrap internal consumer for single topic subscription scene.

Reply:
What happens if this is shared message channel passed in by the application?

Reply:
Usually,  it is better channel closed by producer, not consumer.  When application new a message channel and passes it to singleTopicConsumer/multiTopicConsumer/regexConsumer, we can still close the channel in interface Close() exactly once.